hnn_core/tests/test_drives.py

```diff
@@ -0,0 +1,117 @@
+# Authors: Mainak Jas <mainakjas@gmail.com>
+#          Christopher Bailey <bailey.cj@gmail.com>
+
+import pytest
+
+import os.path as op
+
+import hnn_core
+from hnn_core import Network, read_params
+
+
+def test_add_drives():
+    """Test methods for adding drives to a Network."""
+    hnn_core_root = op.dirname(hnn_core.__file__)
+    params_fname = op.join(hnn_core_root, 'param', 'default.json')
+    params = read_params(params_fname)
+    net = Network(params, legacy_mode=False)
```

Comment:
Is it necessary to have a test with legacy mode?

Reply:
_All_ tests are in 'legacy mode', _except_ this one, as it's testing the new API explicitly.

The most insidious place this is used is in `network.py:L591`:

```python
        if self._legacy_mode:
            # XXX tests must match HNN GUI output
            target_populations = self.cellname_list
```

That's needed to get all the tests against HNN GUI to pass. The code will work fine without, but then there'll be zero chance of `prng` seeds to match those in GUI.

The `net._legacy_mode` also skips parameter checking for values extracted from `param`-files: these can contain all manner of rubbish (notably: drive `tstart`-values that are `>tstop`)